sdk/ml/azure-ai-ml/azure/ai/ml/_artifacts/_artifact_utilities.py

```diff
@@ -359,6 +370,125 @@ def _update_gen2_metadata(name, version, indicator_file, storage_client) -> None
+    data = {
+        "assetId": asset_id,
+        "temporaryDataReferenceId": temporary_data_reference_id,
+        "temporaryDataReferenceType": "TemporaryBlobReference"
+    }
+    data_encoded = json.dumps(data).encode('utf-8')
+    serialized_data = json.loads(data_encoded)
+
+    # make sure correct cloud endpoint is used
+    cloud_endpoint = _get_cloud_details()['registry_discovery_endpoint']
+    service_url = replace_between(cloud_endpoint, HTTPS_PREFIX, ".", workspace.location)
+
+    # send request
+    request_url = f"{service_url}assetstore/v1.0/temporaryDataReference/createOrGet"
+    response = requests_pipeline.post(request_url, json=serialized_data, headers=request_headers)
+
+    if response.status_code != 200:
+        raise HttpResponseError(response=response)
+
+    response_json = json.loads(response.text())
+
+    # get SAS uri for upload and blob uri for asset creation
+    blob_uri = response_json['blobReferenceForConsumption']['blobUri']
+    sas_uri = response_json['blobReferenceForConsumption']['credential']['sasUri']
+
+    return sas_uri, blob_uri
+
+
+def _get_asset_by_hash(
+    operations: "DatastoreOperations",
+    hash_str: str,
+    request_headers: Dict[str, str],
+    workspace: Workspace,
+    requests_pipeline: HttpPipeline) -> Dict[str, str]:
+    """
+    Check if an asset with the same hash already exists in the workspace. If so, return the asset name and version.
+
+    :param operations: Datastore Operations object from MLClient
+    :type operations: azure.ai.ml.operations._datastore_operations.DatastoreOperations
+    :param hash_str: The hash of the specified local upload
+    :type hash_str: str
+    :param request_headers: Request headers for API call
+    :type request_headers: Dict[str, str]
+    :param workspace: Workspace object
+    :type workspace: azure.ai.ml._restclient.v2022_05_01.models.Workspace
+    :param requests_pipeline: Proxy for sending HTTP requests
+    :type requests_pipeline: azure.ai.ml._utils._http_utils.HttpPipeline
+    :return: Existing asset's name and version, if found
+    :rtype: Optional[Dict[str, str]]
+    """
+    existing_asset = {}
+    hash_version = get_content_hash_version()
+
+    # get workspace credentials
+    subscription_id = operations._subscription_id
+    resource_group_name = operations._resource_group_name
+
+    # make sure correct cloud endpoint is used
+    cloud_endpoint = _get_cloud_details()['registry_discovery_endpoint']
+    service_url = replace_between(cloud_endpoint, HTTPS_PREFIX, ".", workspace.location)
+    request_url = f"{service_url}content/v2.0/subscriptions/{subscription_id}/"\
+        f"resourceGroups/{resource_group_name}/providers/Microsoft.MachineLearningServices/workspaces/"\
+        f"{workspace.name}/snapshots/getByHash?hash={hash_str}&hashVersion={hash_version}"
+
+    response = requests_pipeline.get(request_url, headers=request_headers)
+    if response.status_code == 404:
```

Comment:
This is redundant.

Reply:
Which part is redundant?

Reply:
Ah yes - good point. In an earlier version of the code, I had a specific message for 404 error, that's why. I'll remove it now since they're returning the same thing.